### PR TITLE
Fix testcases for cross compilation

### PR DESCRIPTION
**Pull request by catango**
Downgrade bindgen version to 0.70.1 to fix test cases. New bindgen versions break cross build

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- [test] Bindgen 0.71 breaks test cases when crosscompiling. Using bindgen 0.70.1 still works.
```
Comment (photovoltex) [inline CHANGELOG.md:27]: I'm not so sure if it make sense to add a changelog entry for a CI fix. Usually a changelog is intented for updates in relation to the library/binary.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -273,7 +273,9 @@ jobs:
       - name: Fetch
         run: cargo fetch --locked
       - name: Build
-        run: cargo build --frozen --verbose --target ${{ matrix.target }} --no-default-features
+        run: |
+          cargo install bindgen-cli@0.70.1
```
Comment (photovoltex) [inline .github/workflows/test.yml:277]: Adding a separate stage would probably make more sense here. Installing a dependency in a stage named `Fetch` seems not correct in my opinion.

Additionally, why can't we use the latest version of bindgen-cli? And would it maybe make sense to add documentation that bindgen-cli is now required for cross compilation?

**Review by photovoltex — COMMENTED** [review]
Thanks for the fixes. Some thoughts on the topic

File: .github/workflows/test.yml (hunk 2 above)
Comment (photovoltex) [reply .github/workflows/test.yml:277]: From what I read from the docs of `aws-lc-rs` (https://aws.github.io/aws-lc-rs/requirements/linux#bindgen), we seem to need bindgen as we compile for "other" platforms. So installing `clang` and the latest `bindgen-cli` should also work.

Comment (catango) [reply .github/workflows/test.yml:277]: Latest bindgen does not work and bindings cannot be properly generated for the "other" platforms. Did not dig deeper, just downgraded to 0.70.1. Clang is installed by default in Github runner - so no changes needed here.
Will add some documentation and restructuring.

File: CHANGELOG.md (hunk 1 above)
Comment (kingosticks) [reply CHANGELOG.md:27]: I think it's fine to put this here. Sometimes this file is called NEWS, which is what this is. It's a useful thing to know and I can't think of another place more likely to be seen than here.

Is it just cargo-deb that gets broken, rather than cross-compiling in general? I've previously run into a few edge-cases where cargo-deb falls over. I normally now use it with `--no-build` and sort that out myself.

File: .github/workflows/test.yml (hunk 2 above)
Comment (photovoltex) [reply .github/workflows/test.yml:277]: The intressting part is, that we didn't had `bindgen-cli` installed before (at least from what I saw). Which makes one wonder how the pipe even compiled before. 

Anyhow, I also did experiment a bit yesterday and these changes https://github.com/librespot-org/librespot/commit/70974a7bad7fc1ee1da89b54b99796c6d150adf3 also fixed the CI. As seen here (https://github.com/librespot-org/librespot/actions/runs/12735075090/job/35493451448) it does work with the latest version. But I also didn't find out why exactly this works. Might take a look later into it again.

Comment (kingosticks) [reply .github/workflows/test.yml:277]: > The intressting part is, that we didn't had `bindgen-cli` installed before (at least from what I saw). Which makes one wonder how the pipe even compiled before.

I share your wonder. Something must have changed upstream, but I can't see what.

Comment (kingosticks) [reply .github/workflows/test.yml:277]: https://github.com/aws/aws-lc-rs/issues/624 seems kinda related but doesn't change that some of the platforms we're building should have always required bindgen-cli

Maybe it's due to Github's Ubuntu "latest" changing from 22.04 to 24.04  ?

Comment (kingosticks) [reply .github/workflows/test.yml:277]: > Maybe it's due to Github's Ubuntu "latest" changing from 22.04 to 24.04 ?

Yes! https://github.com/actions/runner-images/issues/10636

> Removed from the Ubuntu 24.04 image due to maintenance reasons.

Comment (photovoltex) [reply .github/workflows/test.yml:277]: `apt install bindgen` seems to install a quite old `bindgen` version. Something around 0.59. So it seems necessary to install it via cargo.

Comment (kingosticks) [reply .github/workflows/test.yml:277]: The old 22.04 version has served us well until now. It looks like 0.66 should be in the 24.04 apt repo https://launchpad.net/ubuntu/noble/armhf/bindgen/0.66.1-4 but I don't have an Ubuntu system handy anymore to try. But yeh, taking the latest working version straight from cargo might be better. Maybe we just need to pin that version for the next time they break it.

Comment (photovoltex) [reply .github/workflows/test.yml:277]: As it's a bit tedious to check github actions manuallu, I use [act](https://github.com/nektos/act) locally to test different changes on the workflow. And 24.04 with bindgen installed via apt does seem to work :D

File: CHANGELOG.md (hunk 1 above)
Comment (photovoltex) [reply CHANGELOG.md:27]: Sounds good. It would be nice to have a reference to https://github.com/actions/runner-images/issues/10636 in the comment as we need to install bindgen manually as it was removed from the base image runner.

**Comment by photovoltex** [comment]
@catango friendly bump to address the review comments
